ranges.md

```diff
@@ -0,0 +1,390 @@
+               PRERELEASE versions are being included.
+
+        1. A partial version used with the `<` operator allows any version
+           whose parts of the SemVer (MAJOR,MINOR,PATCH) tuple are
+           less than the specified portions in the partial version.  Thus:
+
+            1. `<I.J` is eqivalent to `<I.J.0-0`
+
+            1. `<I` is eqivalent to `<I.0.0-0`
+
+        1. A partial version used with the `>=` operator allows any version
+           whose parts of the SemVer (MAJOR,MINOR,PATCH) tuple are
+           greater than or equal to the specified portions in the partial
+           version.  Thus:
+
+            1. `>=I.J` is eqivalent to `>=I.J.0`, or `>=I.J.0-0` if
+               PRERELEASE versions are being included.
+
+            1. `>=I` is eqivalent to `>=I.0.0`, or `>=I.0.0-0` if
+               PRERELEASE versions are being included.
+
+        1. A partial version used with the `<=` operator allows any version
+           whose parts of the SemVer (MAJOR,MINOR,PATCH) tuple are
+           less than or equal to the specified portions in the partial
+           version.  Thus:
+
+            1. `<=I.J` is eqivalent to `<I.(J+1).0-0`.
+
+            1. `<=I` is eqivalent to `<(I+1).0.0-0`.
+
+        1. Hyphen ranges, caret ranges, and tilde ranges are expanded as
+           normally with partial versions, then those partial versions are
+           resolved with the relevant operator attached as described above
+           in this section.
+
+1. SemVer Build metadata (that is, a set of identifiers prefixed by `+`) is
+   not considered when evaluating version strings for inclusion in a SemVer
+   Range, as they are not relevant to evaluating precedence.
+
+     Implementations MAY reject Ranges with SemVer strings that include
+     build metadata, but if they do not, then they MUST ignore any build
+     metadata when evaluating versions for inclusion.  For example, the
+     range `>=1.2.3+build.123` MAY be rejected as an invalid Range, but if
+     it is not rejected, then it MUST be equivalent to `>=1.2.3`.
+
+Backus-Naur Form Grammar for Valid SemVer Ranges
+------------------------------------------------
+
+Note: the following grammar _does_ allow the inclusion of build metadata in
+the SemVer Versions used in Comparators.  Implementations MAY reject Ranges
+that include build metadata, but MUST ignore included build metadata if
+allowed.
+
+```bnf
+<semver range> ::= <comparator set>
+                 | <comparator set> <logical or> <range>
+
+<comparator set> ::= <hyphen> | <simple set> | ""
+
+<logical or> ::= <optional space> "||" <optional space>
+
+<hyphen> ::= <partial> <space> "-" <space> <partial>
+
+<simple set> ::= <simple>
+               | <simple> <space> <simple set>
+
+<simple> ::= <primitive> | <partial> | <tilde> | <caret>
+
+<tilde> ::= "~" <partial>
+
+<caret> ::= "^" <partial>
+
+<operator> :: "<" | ">" | ">=" | "<=" | "=" | ""
```

Comment:
Without re-reading the above, its confusing why an empty string is a valid operator. Normally I would suggest the oppose, but perhaps define `<no operator>` as `""` and list that as one of the alternatives, instead of `""` directly, so there is context

Reply:
An empty string is equivalent to using `=`.  For example, depending on `some-module@1.2.3` is equivalent to `some-module@=1.2.3`.

This is important when you have ranges that list individual versions along with other parts.  For example, a vulnerability fixed in version 1.2.3, but then regressed in 1.2.4, and fixed properly in 1.2.5, could express the fixed version set as `1.2.3 || >1.2.4`.